libcnb-package/src/lib.rs

```diff
@@ -106,38 +106,20 @@ fn create_file_symlink<P: AsRef<Path>, Q: AsRef<Path>>(
 ///
 /// # Errors
 ///
-/// Will return an `Err` if any I/O errors happen while walking the file system.
-pub fn find_buildpack_dirs(start_dir: &Path, ignore: &[&Path]) -> std::io::Result<Vec<PathBuf>> {
-    fn find_buildpack_dirs_recursive(
-        path: &Path,
-        ignore: &[&Path],
-        accumulator: &mut Vec<PathBuf>,
-    ) -> std::io::Result<()> {
-        if ignore.contains(&path) {
-            return Ok(());
-        }
-
-        let metadata = path.metadata()?;
-        if metadata.is_dir() {
-            let entries = fs::read_dir(path)?;
-            for entry in entries {
-                let entry = entry?;
-                let metadata = entry.metadata()?;
-                if metadata.is_dir() {
-                    find_buildpack_dirs_recursive(&entry.path(), ignore, accumulator)?;
-                } else if let Some(file_name) = entry.path().file_name() {
-                    if file_name.to_string_lossy() == "buildpack.toml" {
-                        accumulator.push(path.to_path_buf());
-                    }
+/// Will return an `Err` if any I/O errors happen while walking the file system or any parsing errors
+/// from reading a gitignore file.
+pub fn find_buildpack_dirs(start_dir: &Path) -> Result<Vec<PathBuf>, ignore::Error> {
+    let mut buildpack_dirs = vec![];
+    for result in ignore::Walk::new(start_dir) {
```

Comment:
Would writing this using [Iterator::filter_map](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.filter_map) rather than a loop be cleaner? (Iterators can be collected into a `Result`)

Reply:
> (Iterators can be collected into a Result)

and will also fail fast on the first `Err` since iterators are lazy. :tada: